packages/smithy-aws-core/tests/unit/endpoints/test_standard_regional.py

```diff
@@ -0,0 +1,62 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+from smithy_aws_core.endpoints.standard_regional import (
+    StandardRegionalEndpointsResolver,
+    RegionalEndpointParameters,
+)
+
+from smithy_core import URI
+from smithy_http.endpoints import EndpointResolutionError
+
+import pytest
+
+
+async def test_resolve_endpoint_with_valid_sdk_endpoint_string():
+    resolver = StandardRegionalEndpointsResolver(endpoint_prefix="service")
+    params = RegionalEndpointParameters(
+        sdk_endpoint="https://example.com/path?query=123", region=None
+    )
+
+    endpoint = await resolver.resolve_endpoint(params)
+
+    assert endpoint.uri.host == "example.com"
+    assert endpoint.uri.path == "/path"
+    assert endpoint.uri.scheme == "https"
+    assert endpoint.uri.query == "query=123"
+
+
+async def test_resolve_endpoint_with_sdk_endpoint_uri():
+    resolver = StandardRegionalEndpointsResolver(endpoint_prefix="service")
+    parsed_uri = URI(
+        host="example.com", path="/path", scheme="https", query="query=123", port=443
+    )
+    params = RegionalEndpointParameters(sdk_endpoint=parsed_uri, region=None)
+
+    endpoint = await resolver.resolve_endpoint(params)
+
+    assert endpoint.uri == parsed_uri
+
+
+async def test_resolve_endpoint_with_invalid_sdk_endpoint():
+    resolver = StandardRegionalEndpointsResolver(endpoint_prefix="service")
+    params = RegionalEndpointParameters(sdk_endpoint="invalid-uri", region=None)
+
+    with pytest.raises(EndpointResolutionError):
+        await resolver.resolve_endpoint(params)
+
+
+async def test_resolve_endpoint_with_region():
+    resolver = StandardRegionalEndpointsResolver(endpoint_prefix="service")
+    params = RegionalEndpointParameters(sdk_endpoint=None, region="us-west-2")
+
+    endpoint = await resolver.resolve_endpoint(params)
+
+    assert endpoint.uri.host == "service.us-west-2.amazonaws.com"
+
+
+async def test_resolve_endpoint_with_no_sdk_endpoint_or_region():
+    resolver = StandardRegionalEndpointsResolver(endpoint_prefix="service")
+    params = RegionalEndpointParameters(sdk_endpoint=None, region=None)
+
+    with pytest.raises(EndpointResolutionError):
+        await resolver.resolve_endpoint(params)
```

Comment:
There's the inverse of this where both are set that I was gonna ask about in the implementation above. Conceptually I guess this is the same as using an endpoint override for precedence but it may be surprising if the user doesn't realize they set both. A quick debug log above when both are set and having a test ensuring the endpoint is always used is probably useful for catching regressions.

Reply:
Yeah this is a good point.  In our standard endpoint rules we don't raise or log when both the custom endpoint and region are set.  Because region is *also* used as a signing configuration, for sigv4 services, customers often need to specify both (ie, if you are setting custom endpoint you almost always will need to set a region to ensure signing works correctly as well).  

I can definintely add a debug log, but we'll loose that behavior when moving to endpoints2.0.

Either way - good call, will add a test case.

Reply:
Yeah, that's fine to skip the log. We've got the test and that's fine for now. Thanks!